p2p/peering.go

```diff
@@ -1215,6 +1254,98 @@ func (rp *RemotePeer) CFilters(ctx context.Context, blockHashes []*chainhash.Has
 	return filters, nil
 }
 
+// CFilterV2 requests a version 2 regular compact filter from a RemotePeer
+// using getcfilterv2.  The same block can not be requested concurrently from
+// the same peer.
+//
+// The inclusion proof data that ensures the cfilter is committed to in the
+// header is returned as well.
+func (rp *RemotePeer) CFilterV2(ctx context.Context, blockHash *chainhash.Hash) (*gcs2.FilterV2, uint32, []chainhash.Hash, error) {
```

Comment:
this method should verify that the pver is at least wire.CFilterV2Version.  It's probably also a good idea to make it possible to specify the minimum pver during connect and handshake if we know that we will always require the newer version.

Reply:
Ok, added the verification to the specific function and a commit that introduces a minimum version for remote peers. I can refactor that one out into a separate PR if you feel that's more applicable here.